File: CSE331-Algorithm and Complexity/HW4/HW4C++/Driver.cpp
#include "HW4Utility.h"
#include "Solution.cpp"
#include "Graph.h"
#include <fstream>

class Driver;

int main(int argc, char** argv) {
    if(argc != 2) {
        printf("Please provide the input filename as the argument.\n");
        return 0;
    }
    HW4Utility Utility;
    Graph input = Utility.readFile(argv[1]);
    
    Solution s = Solution(input.startNode, input.graph);
    vector<int> studentSolution = s.outputDistances();
    cout << "Your solution " << endl;
    cout << "==========================================" << endl;
    cout << "[";
	for (int i = 0; i < studentSolution.size(); i++)
	{
        if(i == studentSolution.size()-1){
            cout << studentSolution[i];
        }
        else{
            cout << studentSolution[i] << ", ";
        }
        
	}  
    cout << "]" << endl;
    return 0;
}

File: CSE331-Algorithm and Complexity/HW10/src/ub/cse/algo/Driver.java
package ub.cse.algo;

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Arrays;

import java.io.IOException;
import java.io.FileNotFoundException;

/**
 * Class for running the grader. Will take in a command line argument specifying
 * the number of testcases to run.
 */
public class Driver {

    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Please provide the testcase filename as a command line argument");
        }
        String inputFilename = args[0];

        Graph input = read_file(inputFilename);

        Solution student = new Solution(input.getOrigin(),
                                        input.getIncomingEdges(),
                                        input.getOutgoingEdges());
        int[] min_distance = student.outputShortestDistance();

        System.out.println("Your solution:");
        System.out.println("================================================================================");
        System.out.println(Arrays.toString(min_distance));
        System.out.println("================================================================================");
    }


    public static Graph read_file(String inputFilePath){
        int origin = 0;
        ArrayList<HashMap<Integer, Integer>> incoming_edges = new ArrayList<HashMap<Integer, Integer>>();
        ArrayList<HashMap<Integer, Integer>> outgoing_edges = new ArrayList<HashMap<Integer, Integer>>();
        BufferedReader reader = null;

        try {
            reader = new BufferedReader(new FileReader(inputFilePath));
            origin = Integer.valueOf(reader.readLine());
            String line = "";
            while ((line = reader.readLine()) != null) {
                HashMap<Integer, Integer> incoming = new HashMap<>();
                String[] parts = line.split("\\s+");
                for (int i = 0; i < parts.length; i += 2) {
                    incoming.put(Integer.valueOf(parts[i]), Integer.valueOf(parts[i + 1]));
                }
                HashMap<Integer, Integer> outgoing = new HashMap<>();
                parts = reader.readLine().split("\\s+");
                for (int i = 0; i < parts.length; i += 2) {
                    outgoing.put(Integer.valueOf(parts[i]), Integer.valueOf(parts[i + 1]));
                }
                incoming_edges.add(incoming);
                outgoing_edges.add(outgoing);
            }
            reader.close();

          } catch (FileNotFoundException e) {
              System.err.println("Unable to open the file " + inputFilePath);
              e.printStackTrace();
          } catch (IOException e) {
              e.printStackTrace();
          } catch (NullPointerException e) {
              e.printStackTrace();
          }
          return new Graph(origin, incoming_edges, outgoing_edges);
    }
}
